dgv/monitoring/hvd/task_functions.py

```diff
@@ -493,39 +508,58 @@ def update_grist(ti):
                 ["id2", "url", "title"] + columns_to_update,
             ].iloc[0]
         )
-    if not new_rows:
-        return
-    logging.info(f"Adding {len(new_rows)} rows")
-    table.from_dataframe(
-        df=pd.DataFrame(new_rows).rename({"id2": "id"}, axis=1),
-        append="lazy",
-    )
-    ti.xcom_push(
-        key="new_rows",
-        value=[(r["title"], r["url"], r["organization"]) for r in new_rows],
-    )
-    if not removed_hvd:
-        return len(new_rows)
-    to_send = []
-    for hvd_id in removed_hvd:
-        r = requests.get(
-            f"https://www.data.gouv.fr/api/1/datasets/{hvd_id}/",
-            headers={"X-fields": "title,organization{name}"},
+    if new_rows:
+        logging.info(f"Adding {len(new_rows)} rows")
+        table.from_dataframe(
+            df=pd.DataFrame(new_rows).rename({"id2": "id"}, axis=1),
+            append="lazy",
         )
-        r.raise_for_status()
-        r = r.json()
-        to_send.append((hvd_id, r["title"], r["organization"]["name"]))
-    message = ":alert: @clarisse Les jeux de données suivants ont perdu leur tag HVD :"
-    for _id, title, orga in to_send:
-        message += (
-            f"\n- [{title}](https://www.data.gouv.fr/fr/datasets/{_id}/)"
-            f" de l'organisation {orga}"
+        ti.xcom_push(
+            key="new_rows",
+            value=[(r["title"], r["url"], r["organization"]) for r in new_rows],
         )
-        table.update_records(
-            conditions={"id2": _id},
-            new_values={"missing_hvd_tag": True},
+    if removed_hvd:
+        to_send = []
+        for hvd_id in removed_hvd:
+            r = requests.get(
+                f"https://www.data.gouv.fr/api/1/datasets/{hvd_id}/",
+                headers={"X-fields": "title,organization{name}"},
+            )
+            r.raise_for_status()
+            r = r.json()
+            to_send.append((hvd_id, r["title"], r["organization"]["name"]))
+        message = (
+            ":alert: @clarisse Les jeux de données suivants ont perdu leur tag HVD :"
         )
-    send_message(message, MATTERMOST_MODERATION_NOUVEAUTES)
+        for _id, title, orga in to_send:
+            message += (
+                f"\n- [{title}](https://www.data.gouv.fr/fr/datasets/{_id}/)"
+                f" de l'organisation {orga}"
+            )
+            table.update_records(
+                conditions={"id2": _id},
+                new_values={"missing_hvd_tag": True},
+            )
+        send_message(message, MATTERMOST_MODERATION_NOUVEAUTES)
+
+    # updating quality metrics
+    logging.info("Updating datasets quality metrics columns")
+    df = table.to_dataframe(columns_labels=False, usecols=["id2"])
+    df["id2"].apply(update_quality)
+    file_name = "grist_hvd.csv"
+    table.to_dataframe().to_csv(DATADIR + file_name, sep=";", index=False)

```

Comment:
shouldn't we use `df.to_csv()`? Is this going to be used for monitoring? and then don't we want to update a file on minio instead of having one file by date?

Reply:
Yep, I thought that update_quality may update the df itself but nope.

I was wondering for a unique file with month information inside, but let's keep it as it is :pray: